lib/SILGen/SILGenExpr.cpp

```diff
@@ -2987,10 +2987,8 @@ static RValue emitGradientInst(RValueEmitter &RVE, const SGFContext &C,
       break;
     }
     case swift::AutoDiffParameter::Kind::Self: {
-      // `self` is the last parameter of the SIL function.
-      auto selfIdx = origSILTy->getNumParameters() - 1;
-      loweredParamIndices.push_back(selfIdx);
-      break;
+      assert(false &&
```

Comment:
Instead of doing this, the differential operator should use an unsigned index as the parameter, instead of the AutoDiffParameter union.

Reply:
Oops, good point. In addition to an unsigned index, a SourceLoc is also needed, so I'll create a `AutoDiffIndexParameter` wrapper struct.